### PR TITLE
Add a thread count option

### PR DESCRIPTION
**Pull request by gsquire**
This PR adds an option to configure the amount of threads that the directory iterator will use. It doesn't close any issues but rather removes a "TO DO" left in the code.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -444,6 +444,7 @@ impl Fastmod {
 
     fn run_interactive(
         &mut self,
+        threads: &str,
```
Comment (swolchok) [inline src/main.rs]: why not parse this during argument parsing and require a `usize`?

File: src/main.rs [hunk 2]
```diff
@@ -518,13 +520,15 @@ impl Fastmod {
     }
 
     fn run_fast(
+        threads: &str,
```
Comment (swolchok) [inline src/main.rs]: doesn't seem like threads is the most important thing for callers to worry about. 1) why is it the new first argument? 2) why isn't it optional?

**Comment by swolchok** [comment]
What's your use case for specifying the number of threads? I want to make sure someone actually needs this change.

File: src/main.rs (hunk 2 above)
Comment (gsquire) [reply src/main.rs]: I thought this was a private function abstracted away via the CLI. The default would be the number of CPUs which is detected at run time.

**Comment by gsquire** [comment]
> What's your use case for specifying the number of threads?

I don't have a specific one. It was just a patch for the old TODO comment I saw.

**Comment by swolchok** [comment]
First of all, let me apologize for not asking about the use case up front. Sorry about that.

I think that we should table this feature until there is demand for it from users. Thanks for implementing it!

**Pull request closed without merging**